src/Pool.ts

```
@@ -586,29 +587,13 @@ export class Pool<RawResource> {
   destroyAllNow(): Promise<void> {
     this._log('force destroying all objects', 'info');
 
-    const willDie = this._availableObjects.slice();
-    const todo = willDie.length;
-
     this._removeIdleScheduled = false;
     clearTimeout(this._removeIdleTimer);
 
-    // prettier-ignore
-    return new Promise/*:: <void> */((resolve) => {
-      if (todo === 0) {
-        return resolve();
-      }
-
-      let resource;
-      let done = 0;
-
-      while ((resource = willDie.shift())) {
-        this.destroy(resource.resource);
-        ++done;
-
-        if (done === todo && resolve) {
-          return resolve();
-        }
-      }
-    });
+    return Promise.all(
+      this._availableObjects.map((resource) => {
```

`destroy` has a statement like `_count -= 1`, which can have unintended implications when invoked in parallel, I suggest to serially destroy resources

---

When we are doing it serially do we want it to continue if one fails?

---

Calling `destroy` will remove resource from pool even if resource destruction fails, so yeah continue removing objects from pool, use `this._log` to log errors.

I would not recommend swallowing errors, so perhaps return an error with `errors` property, something like [`AggregateError`](https://github.com/sequelize/sequelize/blob/master/lib/errors/aggregate-error.js#L12)